docs/dynamic-component-loading.md

```diff
@@ -0,0 +1,48 @@
+# Dynamic component loading
+
+## Problem statement
+.NET Core runtime has only limited support for dynamically loading assemblies which have additional dependencies or possibly collide with the app in any way. Out of the box only these scenarios really work:
+* Assemblies which have no additional dependencies other than those find in the app itself (`Assembly.Load` and `Assembly.LoadFrom`)
```

Comment:
I do not think that `Assembly.LoadFrom` should be mentioned in the first bullet. It is mentioned in the second bullet already.

Reply:
I was thinking about it... `Assembly.Load` only loads by assembly name, `Assembly.LoadFrom` can do the same but given a file path. But I'll remove it for simplicity.